### PR TITLE
Locale model cleanup

### PR DESCRIPTION
**Pull request by dmk42**
In the numa and knl locale models, `NumaDomain`'s `getChildren()` method was commented out while waiting on a future, `test/functions/iterators/sungeun/iterInClass`.  That test now passes, so this change uncomments the code and provides a test to verify it.

This closes issue #8508 .

Tested on the numa locale model on Mac, Linux, and Cray systems, with C and LLVM back ends.
Tested on the knl locale model on a KNL Cray system with the C back end.
(The LLVM back end has an unrelated problem on a KNL Cray system; see issue #8901 .)

### CODE REVIEW

**Comment by dmk42** [comment]
@gbtitus - Would you please review?  No rush; it's post-release.

File: modules/internal/localeModels/numa/LocaleModel.chpl [hunk 1]
```diff
@@ -85,14 +85,10 @@ module LocaleModel {
...
-    // This is commented out b/c it leads to an internal error during
-    // the resolveIntents pass.  See
-    // test/functions/iterators/sungeun/iterInClass.future
-    //
-    // iter getChildren() : locale {
-    //  halt("No children to iterate over.");
-    //  yield nil;
-    // }
+    iter getChildren() : locale {
+      halt("No children to iterate over.");
```
Comment (cassella) [inline modules/internal/localeModels/numa/LocaleModel.chpl]: Heh.  Somehow I'd managed not to notice that both commented-out functions were just going to halt anyways.

**Comment by gbtitus** [comment]
Looks good!  👍

**Comment by dmk42** [comment]
I don't know how I closed this; I don't remember doing it.  I'm reopening it because it hasn't been merged yet.

**Pull request merged**